src/components/menu/__internal__/submenu/submenu.component.js

```diff
@@ -43,49 +48,52 @@ const Submenu = React.forwardRef(
     },
     ref
   ) => {
-    const [blockDoubleFocus, setBlockDoubleFocus] = useState(false);
+    const submenuRef = useRef(null);
+    const subemenuId = useRef(guid());
     const menuContext = useContext(MenuContext);
-    const {
-      inFullscreenView,
-      openSubmenuIndex,
-      setOpenSubmenuIndex,
-    } = menuContext;
+    const { inFullscreenView, openSubmenuId, setOpenSubmenuId } = menuContext;
     const [submenuOpen, setSubmenuOpen] = useState(false);
-    const [submenuFocusIndex, setSubmenuFocusIndex] = useState(undefined);
+    const [submenuFocusId, setSubmenuFocusId] = useState(null);
+    const [submenuItemIds, setSubmenuItemIds] = useState([]);
     const [characterString, setCharacterString] = useState("");
-    const formattedChildren = React.Children.map(children, (child) => {
-      if (child.type === ScrollableBlock) {
-        const blockChildren = [...child.props.children];
+    const [shiftTabPressed, setShiftTabPressed] = useState(false);
```

Comment:
Again happy to discuss in more detail via slack if you prefer but will give some form of answer here so it's visible to other reviewers. This is needed to allow the input icon to be focused when back tabbing up the submenu items. As an item with a Search can potentially have 2 tabbable elements we need a mechanism for preventing the standard keyboard implementation from overriding the natural tab order here; without it will focus the input every time a user back tabs, ignoring the icon. I'm not a huge fan of it but struggled to find alternatives and took the view that we had similar flags set for `blockDoubleFocus` so I feel this is acceptable in the absence of a better approach. One alternative I did explore was blocking the behaviour in the submenu keydown handler by checking the event target etc. However, whilst it works, I decided against it as it's adding more complexity to the handler and splitting the logic for this behaviour across both submenu and menu item components

Reply:
OK thanks - I still don't like it but it's clear there is a reason for it and you've thought about it, I don't have an immediately better suggestion. About my only one would be to make it a ref, rather than state, as I don't think this needs to cause a rerender?

Reply:
Yeah agree that's probably a better approach 👍